src/core/net/ip6.cpp

```diff
@@ -1035,7 +1035,7 @@ Error Ip6::ProcessReceiveCallback(Message &          aMessage,
             Udp::Header udp;
 
             IgnoreError(aMessage.Read(aMessage.GetOffset(), udp));
-            VerifyOrExit(Get<Udp>().ShouldUsePlatformUdp(udp.GetDestinationPort()), error = kErrorNoRoute);
+            VerifyOrExit(Get<Udp>().IsPortInUse(udp.GetDestinationPort()) == false, error = OT_ERROR_NO_ROUTE);
```

Comment:
I think this will not work with the Border Agent. It listens on `49191` for both Thread interface and platform interfaces and it expect packets from both Thread network and external network.